### DISCUSSION TITLE
How to implement multi-tenant data isolation based on open-source versions.

### DISCUSSION THREAD

**Discussion by lifagui** [warrant-dev/warrant#339] [Q&A]
Hey Team,

We are using your open-source version of Warrant. We operate in a B2B context, and our data needs to be isolated by tenant. This means that Query, Check, Create Warrant, and other operations need to be isolated at the tenant level. Can this be achieved using your open-source version?

Looking forward to your reply. Thank you.

**Comment by akajla09** [comment]
@lifagui - depending on your specific isolation requirements, you could explore a couple of options:

1. Model your authz graph such that each tenant has its own disjoint warrants. Everything exists in the same authz graph/instance but data is logically separated and not related between tenants.
2. If physical isolation of data between tenants is required, you'll likely need to run multiple instances of the warrant service, one for each tenant.
